src/OrleansProviders/Streams/Common/PooledCache/ICacheDataAdapter.cs

```diff
@@ -59,6 +60,24 @@ public interface ICacheDataAdapter<in TQueueMessage, TCachedMessage>
         /// Assignable purge action.  This is called when a purge request is triggered.
         /// </summary>
         Action<IDisposable> PurgeAction { set; }
+
+        /// <summary>
+        /// Handler for when a cachedMessage got purged
+        /// </summary>
+        /// <param name="cachedMessage"></param>
+        void SignalMessagePurged(ref TCachedMessage cachedMessage);
+
+        /// <summary>
+        /// Determine action to make on a purgeRequest
+        /// </summary>
+        /// <param name="purgeRequest"></param>
+        void ConsultPurgeRequest(IDisposable purgeRequest);
+
+        /// <summary>
+        /// CachePressureMonitor used on its cache
+        /// </summary>
+        ICachePressureMonitor CachePressureMonitor { set; }
```

Comment:
yes, I definitely considered encapsulate those purge related methods into one class.

but the Typed param in ICacheDataAdapter is not making this direction super pretty. But I can definitely improve this by thinking of other alternatives or improvements.